src/main/kotlin/spark/Routes.kt

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2017 Love Löfdahl
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package spark
+
+open class RequestHandler {
+    val start: Long = System.currentTimeMillis()
+
+    fun requestTime(): Long {
+        return System.currentTimeMillis() - start
+    }
+}
+
+class RouteHandler(val request: Request, val response: Response) : RequestHandler()
```

Comment:
add methods for direct implicit access of request and response methods that are meaningful, e.g.:
```java
    params(":name")
    queryParams("q")
    body()
    ...
    etc
```